### DISCUSSION TITLE
"ItemsSoure" updates the data back to the top, is there any good way?

### DISCUSSION THREAD

**Discussion by wosiyourfather** [AvaloniaUI/Avalonia#12694] [Q&A] [answered]
"ItemsSoure" updates the data back to the top, is there any good way?
![up](https://github.com/AvaloniaUI/Avalonia/assets/53590491/4a223537-6379-49e8-a28b-08dca6d32f60)
![data](https://github.com/AvaloniaUI/Avalonia/assets/53590491/0fa72e55-7de7-4a6d-83b3-0f1c4052fb32)

**Comment by timunie** [comment]
use ObservableCollection. Don't update the collection in code behind.

**Reply by wosiyourfather** [reply]
I have a sample that needs to refresh data every second, but as soon as it is refreshed, it returns to the top. Do you have time to help me take a look

**Reply by wosiyourfather** [reply]
[AvaloniaDatagridBindingToDatatable(2).zip](https://github.com/AvaloniaUI/Avalonia/files/12460971/AvaloniaDatagridBindingToDatatable.2.zip)

**Comment by timunie** [comment, marked as answer]
@wosiyourfather please get yourself some time and do the provided tutorials [here](https://github.com/AvaloniaUI/Avalonia.Samples) and [here](https://docs.avaloniaui.net/docs/next/tutorials/)
